x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/TokenAuthIntegTests.java

```diff
@@ -425,8 +425,12 @@ public void testRefreshingMultipleTimesWithinWindowSucceeds() throws Exception {
                     return;
                 }
                 threadSecurityClient.refreshToken(refreshRequest, ActionListener.wrap(result -> {
-                    accessTokens.add(result.getTokenString());
-                    refreshTokens.add(result.getRefreshToken());
+                    if (accessTokens.get().size() > 0 && accessTokens.get().contains(result.getTokenString()) == false ||
+                        refreshTokens.get().size() > 0 && refreshTokens.get().contains(result.getRefreshToken()) == false) {
+                        newToken.set(true);
+                    }
```

Comment:
To play it cool, discard responses (don't fail the test, but log something) if `t2 - t1` is greater than 30 sec.